document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/mapping/mappers/exemptions/ApiToExemptionsMapper.java

```diff
@@ -1,13 +1,78 @@
 package uk.gov.companieshouse.document.generator.company.report.mapping.mappers.exemptions;
 
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
 import org.springframework.web.context.annotation.RequestScope;
+import uk.gov.companieshouse.api.model.exemptions.ExemptionItemsApi;
 import uk.gov.companieshouse.api.model.exemptions.ExemptionsApi;
 import uk.gov.companieshouse.document.generator.company.report.mapping.model.document.items.exemptions.Exemptions;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @RequestScope
 @Mapper(componentModel = "spring", uses = {ApiToExemption.class})
-public interface ApiToExemptionsMapper {
+public abstract class  ApiToExemptionsMapper {
+
+    public abstract Exemptions apiToExemptionsMapper(ExemptionsApi exemptionsApi);
+
+    @AfterMapping
+    void checkActiveExemption(ExemptionsApi exemptionsApi, @MappingTarget Exemptions exemptions) {
+
+        List<ExemptionItemsApi> disclosureTransparencyRulesChapterFiveAppliesList = new ArrayList<>();
+        List<ExemptionItemsApi> pscExemptAsSharesAdmittedOnMarketList = new ArrayList<>();
+        List<ExemptionItemsApi> pscExemptAsTradingOnRegulatedMarketList = new ArrayList<>();
+        List<ExemptionItemsApi> pscExemptAsTradingOnUkRegulatedMarketList = new ArrayList<>();
+
+        if(exemptionsApi.getDisclosureTransparencyRulesChapterFiveApplies() != null) {
+            disclosureTransparencyRulesChapterFiveAppliesList = exemptionsApi.getDisclosureTransparencyRulesChapterFiveApplies().getItems();
+
+            exemptions.setActiveExemption(false);
+
+            for (int i = 0; i < disclosureTransparencyRulesChapterFiveAppliesList.size(); i++) {
+                if (disclosureTransparencyRulesChapterFiveAppliesList.get(i).getExemptTo() == null) {
+                    exemptions.setActiveExemption(true);
+                }
+            }
```

Comment:
This can be wrote as: 

`            for (ExemptionItemsApi exemptionItems : disclosureTransparencyRulesChapterFiveAppliesList) {
                if (exemptionItems.getExemptTo() == null) {
                    exemptions.setActiveExemption(true);
                }
            }
`